src/gc/CMakeLists.txt

```diff
@@ -26,12 +16,48 @@ set( GC_SOURCES
   gceewks.cpp
   handletablecache.cpp)
 
-if(NOT CLR_CMAKE_PLATFORM_UNIX)
-set ( GC_SOURCES
-  ${GC_SOURCES}
-  windows/gcenv.windows.cpp)
-endif(NOT CLR_CMAKE_PLATFORM_UNIX)
+if(CLR_CMAKE_PLATFORM_UNIX)
+  include(unix/configure.cmake)
+  set ( GC_SOURCES
+    ${GC_SOURCES}
+    unix/gcenv.unix.cpp
+    unix/events.cpp
+    unix/cgroup.cpp)
+else()
+  set ( GC_SOURCES
+    ${GC_SOURCES}
+    windows/gcenv.windows.cpp)
+endif(CLR_CMAKE_PLATFORM_UNIX)
+
+if(WIN32)
+  set (GC_LINK_LIBRARIES
+    ${STATIC_MT_CRT_LIB}
+    ${STATIC_MT_VCRT_LIB}
+    kernel32.lib)
+else()
+  set (GC_LINK_LIBRARIES)
+endif(WIN32)
 
 convert_to_absolute_path(GC_SOURCES ${GC_SOURCES})
 
-add_library_clr(gc_standalone STATIC ${GC_SOURCES})
+add_library(gc SHARED ${GC_SOURCES})
```

Comment:
So this is a problem because the function `add_library_clr` is defined after `src\gc\CMakeLists.txt` is included. Looking at the body, it looks like it should be OK to move it to `functions.cmake`. I'll give that a try.